File: android/style1/src/main/java/io/github/aidenkoog/style1/activity/SplashActivity.java
package io.github.aidenkoog.style1.activity;

import android.content.Intent;
import android.os.Bundle;
import androidx.appcompat.app.AppCompatActivity;


public class SplashActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(null);
    }

    @Override
    protected void onNewIntent(Intent intent) {
        super.onNewIntent(intent);
        if (intent != null) {
            setIntent(intent);
        }
    }

    @Override
    protected void onResume() {
        super.onResume();
    }
}

File: android/app/src/main/java/io/github/aidenkoog/android/presentation/splash/SplashActivity.kt
package io.github.aidenkoog.android.presentation.splash

import android.os.Bundle
import android.util.Log
import androidx.activity.viewModels
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import dagger.hilt.android.AndroidEntryPoint
import io.github.aidenkoog.android.R
import io.github.aidenkoog.android.databinding.SplashMainBinding

@AndroidEntryPoint
class SplashActivity : AppCompatActivity() {

    private val viewModel: SplashViewModel by viewModels()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val splashMainBinding =
            DataBindingUtil.setContentView<SplashMainBinding>(this, R.layout.splash_main);
        splashMainBinding.splashViewModel = viewModel

        with(viewModel) {
            loadAppInfo()
            isLoaded.observe(this@SplashActivity, {
                Log.d(SplashActivity::class.simpleName, "loaded!");
            })
            appInfoListLiveData.observe(this@SplashActivity, {
                it.let {
                    Log.d(SplashActivity::class.simpleName, "appInfo data exists!");
                }
            })
        }
    }
}